### PR TITLE
Skip van Vleck corrections based on new FITS-IDI keyword

### PR DESCRIPTION
**Pull request by kettenis**
As described in the VLBA Sensitivity Upgrade Memo 52, new versions of the DiFX correlator will apply the van Vleck when requested to do so.  This will be signalled by a new keyword in the FITS header of FITS-IDI output.  See

  https://library.nrao.edu/public/memos/vlba/up/VLBASU_52.pdf

for more details.

Recognize this new keyword and skip the van Vleck correction if it is set to a non-zero value.

### CODE REVIEW

File: msfits/MSFits/FitsIDItoMS.h
Comment (tammojan) [inline msfits/MSFits/FitsIDItoMS.h:133]: When adding a new argument with a default value, I think it's better to put it at the end.

File: msfits/MSFits/FitsIDItoMS.cc
Comment (tammojan) [inline msfits/MSFits/FitsIDItoMS.cc:2255]: Isn't exact comparison with a floating point number a bit scary?

File: msfits/MSFits/MSFitsIDI.cc
Comment (tammojan) [inline msfits/MSFits/MSFitsIDI.cc:295]: Style: I'd prefer `0.`;

**Review by tammojan — COMMENTED** [review]
Looks good to me, but I'd appreciate it if @dpetry could have a look as well.

File: msfits/MSFits/FitsIDItoMS.h
Comment (kettenis) [reply msfits/MSFits/FitsIDItoMS.h:133]: I can see your point. Conceptually the new vanVleck parameter is related to the existing correlat parameter, so that is why I added it there.  But I can change it.

This actually relates to what I consider to be a bit of an issue with how the code is structure. I think that for all practical purpose the FITSIDItoMS1 class should be considered an "internal" API. Users should be using the MSFitsIDI class (and this is what importfitsidi does). But there is nothing in the documentation that spells this out.

File: msfits/MSFits/MSFitsIDI.cc
Comment (kettenis) [reply msfits/MSFits/MSFitsIDI.cc:295]: I'll fix that.

File: msfits/MSFits/FitsIDItoMS.cc
Comment (kettenis) [reply msfits/MSFits/FitsIDItoMS.cc:2255]: Comparison against 0.0 is perfectly safe as zero can be represented exactly. And since the memo referenced in this PR documents non-zero values as doing something different, this is exactly what is needed.

The habit if representing what should clearly be integer parameters as floats in FITS-related things is weird but that's not something we can change.

File: msfits/MSFits/FitsIDItoMS.h
Comment (tammojan) [reply msfits/MSFits/FitsIDItoMS.h:133]: Given that there is no documentation (either about the parameters, or about the fact that this should be an internal API), could you move the parameter to the end to avoid bugs?

File: msfits/MSFits/MSFitsIDI.cc
Comment (kettenis) [reply msfits/MSFits/MSFitsIDI.cc]: Done

**Review by tammojan — APPROVED** [review]
Looks good to me; let me know if you want to wait for @dpetry's approval.

**Comment by kettenis** [comment]
Let's move on with this. I have the blessing from the NRAO/VLBA folks to move ahead with this.

**Pull request merged**
